app/src/androidTest/java/com/github/se/project/ui/End2End/TutorEndToEnd.kt

```diff
@@ -0,0 +1,124 @@
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.assertTextEquals
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithContentDescription
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performGesture
+import androidx.compose.ui.test.performTextInput
+import androidx.compose.ui.test.swipeRight
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.se.project.PocketTutorApp
+import com.github.se.project.model.lesson.LessonRepository
+import com.github.se.project.model.lesson.LessonViewModel
+import com.github.se.project.model.profile.ListProfilesViewModel
+import com.github.se.project.model.profile.ProfilesRepository
+import com.github.se.project.model.profile.Role
+import com.github.se.project.ui.navigation.NavigationActions
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.any
+import org.mockito.kotlin.whenever
+
+@RunWith(AndroidJUnit4::class)
+class EndToEndTest {
+
+  @Mock lateinit var navigationActions: NavigationActions
+
+  @Mock lateinit var context: Context
+
+  // Mock du ProfilesRepository
+  private val mockProfileRepository = mock(ProfilesRepository::class.java)
+
+  private val mockProfileViewModel = ListProfilesViewModel(mockProfileRepository)
+
+  private val mockLessonRepository = mock(LessonRepository::class.java)
+
+  private val mockLessonViewModel = LessonViewModel(mockLessonRepository)
+
+  // private var currentProfile : Profile = Profile(uid = "1", googleUid = "googleUid", firstName =
+  // "First", lastName = "Last", phoneNumber = "1234567890", role = Role.STUDENT, section =
+  // Section.GM, academicLevel = AcademicLevel.BA1, languages = listOf(), subjects = listOf(),
+  // schedule = listOf())
+  // var currentProfile: Profile? = null
+
+  private val mockUid = "mockUid"
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    context = mock(Context::class.java)
+    whenever(mockProfileRepository.addProfile(any(), any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    whenever(mockProfileRepository.getNewUid()).thenReturn("mockUid")
+  }
+
+  @Test
+  fun TutorEndToEndTest() {
+    composeTestRule.setContent {
+      PocketTutorApp(true, viewModel(), mockProfileViewModel, mockLessonViewModel)
+    }
+    // Sign In Screen
+    composeTestRule.onNodeWithTag("logo").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("loginButton").performClick()
+    composeTestRule.onNodeWithTag("firstNameField").assertIsDisplayed()
+
+    // Create Profile Screen
```

Comment:
These screen-description comments are very helpful! Makes the test a lot more readable